Mathlib/Algebra/Group/Units/Hom.lean

```diff
@@ -518,3 +526,34 @@ protected theorem div_div_cancel_left (h : IsUnit a) : a / b / a = b⁻¹ := by
 end DivisionCommMonoid
 
 end IsUnit
+
+section IsLocalRingHom
+
+variable {G R S T F : Type*}
+
+/-- A local ring homomorphism is a homomorphism `f` between monoids such that `a` in the domain
+  is a unit if `f a` is a unit for any `a`. See `LocalRing.local_hom_TFAE` for other equivalent
+  definitions in the local ring case - from where this concept originates, but it is useful in
+  other contexts, so we allow this generalisation in mathlib. -/
+class IsLocalRingHom [Monoid R] [Monoid S] [MonoidHomClass F R S] (f : F) : Prop where
```

Comment:
I suggest renaming this to `IsLocalHom` or `MonoidHom.IsLocal`.

Reply:
I shall use the name `IsLocalHom`.